drivers/i3c/i3c_mcux.c

```diff
@@ -0,0 +1,1108 @@
+	.ibiCallback = COND_CODE_1(CONFIG_I3C_USE_IBI,
+				   (mcux_i3c_ibi_cb),
+				   (NULL)),
+	.transferComplete = mcux_i3c_ctrl_transfer_callback,
+};
+
+/**
+ * @brief Do RSTDAA to reset dynamic addresses of all devices on bus.
+ *
+ * @param dev Pointer to the device driver instance.
+ */
+static void mcux_i3c_do_rstact_peripheral_all(const struct device *dev)
+{
+	const struct mcux_i3c_config *config = dev->config;
+	I3C_Type *base = config->base;
+
+	LOG_DBG("Broadcast RSTACT");
+
+	/*
+	 * Note that we ignore error from RSTACT as there may not be
+	 * any connected devices responding to this CCC. So we clear
+	 * any error bits anyway.
+	 */
+	if (i3c_ccc_do_rstact_peripheral_all(dev) != 0U) {
+		LOG_DBG("Broadcast RSTACT was NACK.");
+	}
+
+	I3C_MasterClearErrorStatusFlags(base, kI3C_MasterAllErrorFlags);
+}
+
+/**
+ * @brief Do RSTDAA to reset dynamic addresses of all devices on bus.
+ *
+ * @param dev Pointer to the device driver instance.
+ */
+static void mcux_i3c_do_rstdaa_all(const struct device *dev)
+{
+	const struct mcux_i3c_config *config = dev->config;
+	I3C_Type *base = config->base;
+
+	LOG_DBG("Broadcast RSTDAA");
+
+	/*
+	 * Note that we ignore error from RSTDAA as there may not be
+	 * any connected devices responding to this CCC (e.g. no
+	 * device has already been assigned an address, so no need
+	 * to reset). So we clear any error bits anyway.
+	 */
+	if (i3c_ccc_do_rstdaa_all(dev) != 0U) {
+		LOG_DBG("Broadcast RSTDAA was NACK.");
+	}
+
+	I3C_MasterClearErrorStatusFlags(base, kI3C_MasterAllErrorFlags);
+}
+
+/**
+ * @brief Do SETDASA to set static address as dynamic address.
+ *
+ * @param dev Pointer to the device driver instance.
+ * @param[out] True if DAA is still needed. False if all registered
+ *             devices have static addresses.
+ *
+ * @retval 0 if successful.
+ */
+static int mcux_i3c_do_setdasa(const struct device *dev, bool *need_daa)
+{
+	const struct mcux_i3c_config *config = dev->config;
+	struct mcux_i3c_data *data = dev->data;
+	struct i3c_dev_list *dev_list = &data->device_list;
+	I3C_Type *base = config->base;
+
+	struct i3c_ccc_payload ccc_payload;
+	struct i3c_ccc_target_payload ccc_tgt_payload;
+	uint8_t dyn_addr;
+	sys_snode_t *node;
+	int ret;
+
+	*need_daa = false;
+
+	/* Set up common fields for CCC structs */
+	ccc_tgt_payload.rnw = 0;
+	ccc_tgt_payload.data = &dyn_addr;
+	ccc_tgt_payload.data_len = 1;
+
+	memset(&ccc_payload, 0, sizeof(ccc_payload));
+	ccc_payload.ccc.id = I3C_CCC_SETDASA;
+	ccc_payload.targets.payloads = &ccc_tgt_payload;
+	ccc_payload.targets.num_targets = 1;
+
+	/* Loop through the registered I3C devices */
+	SYS_SLIST_FOR_EACH_NODE(&dev_list->devices.i3c, node) {
+		struct i3c_device_desc *desc = (void *)node;
+		struct i3c_ccc_getbcr bcr;
+		struct i3c_ccc_getdcr dcr;
+
+		/*
+		 * A device without static address => need to do
+		 * dynamic address assignment.
+		 */
+		if (desc->static_addr == 0U) {
```

Comment:
If multiple devices (more than 1) don't have a static address. 

I just tried adding it into your fork on to mimxrt685_evk_cm33.overlay and running west build -p -b mimxrt685_evk_cm33 samples/sensor/lsm6dso_i3c/
Looking at the devicetree_unfixed.h, it's only generating the latter of the ones with @0
```
    lsm6dso@0 {
        label = "LSM6DSO";
        compatible = "st,lsm6dso";
        reg = <0x00 0x0208 0x006c100c>;

        status = "okay";
    };

    lsm6dso@0 {
        label = "LSM6DSO";
        compatible = "st,lsm6dso";
        reg = <0x00 0x0208 0x006c100d>;

        status = "okay";
    };
```

I've seen Linux define them like this, where the pid is used within there... that'll probably require more python-devicetree modification :( unless if can get more time to think about this
```
sensor1: sensor@0,39200154004 {
    reg = <0x0 0x392 0x154004>;
};

sensor2: sensor@0,39200179862 {
    reg = <0x0 0x392 0x179862>;
};
```



The workaround I've been doing with a i3c api is to just change a i3c-device@0 to i3c-device@1.. there will be a warning about how reg and unit-address don't match... but it's just a warning 🤪  and will continue to build

Reply:
ok, it might be good to have a note in the `i3c.rst` about that

Reply:
Updated doc.